flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -4801,7 +4801,10 @@ class FMCMainDisplay extends BaseAirliners {
      */
     //TODO: Can this be util?
     getGW() {
-        return (SimVar.GetSimVarValue("TOTAL WEIGHT", "Pounds") * 0.4535934) / 1000;
+        const fuelWeight = SimVar.GetSimVarValue("FUEL TOTAL QUANTITY WEIGHT", getUserUnit());
+        const emptyWeight = SimVar.GetSimVarValue("EMPTY WEIGHT", getUserUnit());
+        const payloadWeight = getTotalPayload();
+        return (emptyWeight + fuelWeight + payloadWeight) / 1000;
```

Comment:
GW should be set equal to the ZFW input + block fuel input by the pilot on the INIT A page, regardless of whether the ZFW is typed in by the pilot, or input by double-clicking the LSK key to automatically load the current ZFW. It should not be based on independently adding the total payload to the empty weight. If the pilot happens to insert the incorrect ZFW and/or block fuel, the weight shown on the ECAM and the weight used for flight path predictions and CDU/PFD display of operating speeds S, F, green dot, and VLS  should be based on these incorrect weights (even though the airplane's flight characteristics/performance will reflect the actual gross weight). Until the ZFW and block fuel are entered, there is no GW value for displaying in the ECAM or to use for flight path or speed predictions.

Once the first engine has started, the fuel weight is then taken from the fuel quantity indicating system instead of the block fuel entered by the pilot. So if the block fuel entered into the MCDU by the pilot is incorrect, the gross weight will be incorrect until the first engine is started.

Reply:
Thanks for the detailed refs, Don. I was looking exactly for this. Will make the changes to the logic

Reply:
Also, although this is getting beyond your initial intention, if you want to add another weight initialization related thing, if the ZFW, ZFWCG, or Block fuel inputs are blank after the first engine is started, the MCDU scratchpad should display the message "INITIALIZE WEIGHT/CG."

Reply:
Here are all the references:
![Screenshot (2110)](https://user-images.githubusercontent.com/70166617/173247018-235bea77-9f5d-4c25-853b-7897b618378d.png)
![Screenshot (2111)](https://user-images.githubusercontent.com/70166617/173247023-f8112c29-5f8e-4020-b384-6e56b168954c.png)
![Screenshot (2112)](https://user-images.githubusercontent.com/70166617/173247036-0065e3a4-f482-4f44-8f4c-c86773235571.png)
![Screenshot (2113)](https://user-images.githubusercontent.com/70166617/173247039-523b790b-c108-410d-8452-b73aace876e8.png)
![Screenshot (2114)](https://user-images.githubusercontent.com/70166617/173247047-73e2f8e9-a508-406f-8432-ffa7e41b5f7e.png)
![Screenshot (2115)_LI](https://user-images.githubusercontent.com/70166617/173247056-0e423959-6120-4e9d-8f95-9e80d0350947.jpg)
![Screenshot (2116)](https://user-images.githubusercontent.com/70166617/173247064-1e9298ab-aac9-41d4-a08a-a0fbed382777.png)

Reply:
@donstim @tracernz 
this one should now be ready for a re-review